src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -660,16 +660,17 @@ public void removeElementFilesAndLangKeys(ModElement element) {
 			int templateID = 0;
 			for (Object list : templateLists) {
 				Map<GeneratorTemplate, List<Boolean>> files = new HashMap<>();
-				String listName = (String) ((Map<?, ?>) list).get("name");
+				String groupName = (String) Objects.requireNonNullElse(((Map<?, ?>) list).get("name"),
+						"Templates group");
 				Object listData = TemplateExpressionParser.processFTLExpression(this,
 						(String) ((Map<?, ?>) list).get("listData"), generatableElement);
 				List<?> templates = (List<?>) ((Map<?, ?>) list).get("forEach");
 				List<?> elementsData = new ArrayList<>();
+				if (listData instanceof Map<?, ?> listMap)
+					elementsData = new ArrayList<>(listMap.entrySet());
+				else if (listData instanceof Collection<?> collection)
+					elementsData = new ArrayList<>(collection);

```

Comment:
(TODO) In case we switch to J18 (J17-preview) before this PR is merged:
```suggestion
				List<?> elementsData = switch (listData) {
					case Map<?, ?> listMap -> new ArrayList<>(listMap.entrySet());
					case Collection<?> collection -> new ArrayList<>(collection);
					case null, default -> new ArrayList<>();
				};
```